PxWeb/Controllers/Api2/Admin/DatabaseController.cs

```diff
@@ -23,56 +24,74 @@ public class DatabaseController : ControllerBase
         {
-            try
+            if (_dataSource.GetType() != typeof(PxFileDataSource))
             {
-                if (_dataSource.GetType() != typeof(PxFileDataSource))
+                return StatusCode(405, "Only possible to generate database for PX-file databases");
+            }
+
+            _backgroundWorkerQueue.QueueBackgroundWorkItem(async token =>
+            {
+                try
                 {
-                    return StatusCode(405, "Only possible to generate database for PX-file databases");
-                }
+                    PXWeb.Database.DatabaseSpider spider;
+                    spider = new PXWeb.Database.DatabaseSpider();
+                    spider.Handles.Add(new AliasFileHandler(_configOptions, _logger));
+                    spider.Handles.Add(new LinkFileHandler(_configOptions, _logger));
+                    spider.Handles.Add(new PxFileHandler());
+                    spider.Handles.Add(new MenuSortFileHandler(_configOptions, _logger));
 
-                PXWeb.Database.DatabaseSpider spider;
-                spider = new PXWeb.Database.DatabaseSpider();
-                spider.Handles.Add(new AliasFileHandler(_configOptions, _logger));
-                spider.Handles.Add(new LinkFileHandler(_configOptions, _logger));
-                spider.Handles.Add(new PxFileHandler());
-                spider.Handles.Add(new MenuSortFileHandler(_configOptions, _logger));
+                    List<string> langs = new List<string>();
+                    foreach (Language lang in _configOptions.Languages)
+                    {
+                        langs.Add(lang.Id);
+                    }
 
-                List<string> langs = new List<string>();
-                foreach (Language lang in _configOptions.Languages)
+                    string sorting = GetSorting(sortOrder);
+                    string databasePath = Path.Combine(_hostingEnvironment.RootPath, "Database");
+
+                    spider.Builders.Add(new MenuBuilder(_configOptions, _logger, _hostingEnvironment, langs.ToArray(), GetLangDependent(langDependent)) { SortOrder = GetSortOrder(sorting) });
+                    spider.Search(databasePath);
+                }
+                catch (System.Exception ex)
                 {
-                    langs.Add(lang.Id);
+                    _responseState.AddEvent(new Event("Error", ex.Message));
+                    _logger.LogError(ex.Message);
                 }
+            });
 
-                string sorting = GetSorting(sortOrder);
-                string databasePath = Path.Combine(_hostingEnvironment.RootPath, "Database");
-
-                spider.Builders.Add(new MenuBuilder(_configOptions, _logger, _hostingEnvironment, langs.ToArray(), GetLangDependent(langDependent)) { SortOrder = GetSortOrder(sorting) });
-                spider.Search(databasePath);
+            return new AcceptedResult();
+        }
 
-                return Ok(spider.Messages);
```

Comment:
Could we mofify the DatabaseSpider project so that it gets a ResponseState injected and writes is build messages to that.